### PR TITLE
Fix out-of-tree build for cmake

### PR DESCRIPTION
**Pull request by lofcek**
File well_known_types_embed.cc is generated in the same place where are source code and this does not allow us make our sources read-only

### CODE REVIEW

File: cmake/libprotoc.cmake [hunk 1]
```diff
@@ -92,6 +91,7 @@ set(libprotoc_files
   ${protobuf_source_dir}/src/google/protobuf/compiler/ruby/ruby_generator.cc
   ${protobuf_source_dir}/src/google/protobuf/compiler/subprocess.cc
   ${protobuf_source_dir}/src/google/protobuf/compiler/zip_writer.cc
+  ${protobuf_BINARY_DIR}/well_known_types_embed.cc		# This file is generated, therefore is in different directory
```
Comment (xfxyjwf) [inline cmake/libprotoc.cmake]: This file list here is generated by the update_file_lists.sh script:
https://github.com/google/protobuf/blob/master/update_file_lists.sh

So your change here won't stand unless you make a corresponding change to src/Makefile.am. Can you make a change to src/Makefile.am accordingly?

Comment (lofcek) [reply cmake/libprotoc.cmake]: I will try, although I have no experience with automake .. but for now I have

Comment (lofcek) [reply cmake/libprotoc.cmake]: something that could fix it for a few days

File: update_file_lists.sh [hunk 2]
```diff
@@ -96,7 +96,12 @@ set_cmake_value() {
       print \$0;
       len = split(values, vlist, \" \");
       for (i = 1; i <= len; ++i) {
-        printf(\"  %s%s\\n\", prefix, vlist[i]);
+	    if (vlist[i] != \"google/protobuf/compiler/js/well_known_types_embed.cc\") {
```
Comment (xfxyjwf) [inline update_file_lists.sh]: Format these using two-spaces indentation.

File: update_file_lists.sh [hunk 3]
```diff
@@ -96,7 +96,12 @@ set_cmake_value() {
...
+           printf(\"  %s%s\\n\", prefix, vlist[i]);
+		} else {
+           printf(\"  \${protobuf_BINARY_DIR}/well_known_types_embed.cc\t\t# This file is generated, therefore is in different directory\\n\");
```
Comment (xfxyjwf) [inline update_file_lists.sh]: Put the comment before the print line.

**Comment by BSBandme** [comment]
Please fix the conflicts first.

**Comment by xfxyjwf** [comment]
This should no longer be needed because I have deleted the problematic js_embed binary in our build.

**Pull request closed without merging**